### PR TITLE
Interior left nav open state

### PR DESCRIPTION
**Pull request by pfarkya**
Closes carbon-design-system/carbon-components-react#715

Adding a new props `initialOpenState` to interiorLeftNav to control it's initial state of open.
It is required when we have smaller screen and the left Navigation is open by default it should be close by default for better UX. 

#### Changelog

**New**

* Added new props `initialOpenState` to interiorLeftNav Component

**Changed**

 interiorLeftNav Component

**Removed**

Removed Nothing

### CODE REVIEW

**Comment by pfarkya** [comment]
@asudoh Thanks for the review comment. I am now updating props in `cWRP`.

File: src/components/InteriorLeftNav/InteriorLeftNav.js [hunk 1]
```diff
@@ -13,22 +13,30 @@ export default class InteriorLeftNav extends Component {
...
+      this.props.initialOpenState !== undefined
+        ? this.props.initialOpenState
+        : true,
   };
 
   componentWillReceiveProps = nextProps => {
     if (nextProps.activeHref) {
       this.setState({ activeHref: nextProps.activeHref });
     }
+    if (nextProps.initialOpenState !== undefined) {
```
Comment (asudoh) [inline src/components/InteriorLeftNav/InteriorLeftNav.js]: The condition should be `this.props.open !== nextProps.open`, meaning, there is the change in the corresponding prop.

**Comment by pfarkya** [comment]
@asudoh changed the prop name from initialOpenState to simply open. Thanks

**Comment by asudoh** [comment]
Thanks @pfarkya! Please introduce `this.props.open !== nextProps.open` check, too.

**Comment by pfarkya** [comment]
@asudoh Changed as requested. Thanks!

**Comment by pfarkya** [comment]
@asudoh When I will expect it to be merged?

File: src/components/InteriorLeftNav/InteriorLeftNav-test.js [hunk 2]
```diff
@@ -46,6 +46,22 @@ describe('InteriorLeftNav', () => {
           .matchesElement(<div data-child-div>foo</div>)
       ).toBe(true);
     });
+
+    it('renders a interior left nav with initialOpenState to be true', () => {
```
Comment (marijohannessen) [inline src/components/InteriorLeftNav/InteriorLeftNav-test.js]: can you change the test text to reflect the prop name change from `initialOpenState` to `open`?

File: src/components/InteriorLeftNav/InteriorLeftNav-test.js [hunk 3]
```diff
@@ -46,6 +46,22 @@ describe('InteriorLeftNav', () => {
...
+      const interiorLeftNav = mount(<InteriorLeftNav open={true} />);
+      expect(interiorLeftNav.length).toEqual(1);
+      expect(
+        interiorLeftNav.find('.bx--interior-left-nav--collapsed').length
+      ).toEqual(0);
+    });
+
+    it('renders a interior left nav with initialOpenState to be false', () => {
```
Comment (marijohannessen) [inline src/components/InteriorLeftNav/InteriorLeftNav-test.js]: Same comment as above

**Comment by pfarkya** [comment]
@marijohannessen changed as reviewed. Thanks!

**Pull request merged**